### ISSUE TITLE
Evaluation Method

### ISSUE THREAD

**Issue by rabitwhte** [RandolphVI/Hierarchical-Multi-Label-Text-Classification#6]
Hi @RandolphVI,

correct me if I'm wrong, but it seems like the evaluation method here is including prediction at all levels, not only at the final one - thus I may have a classifier which does not provide any correct prediction at level 4-th (which is the only one I'm interested in) while at the same time it shows validation accuracy to be 75 % (because it gives perfect match at level 1, 2 and 3). Is that the case?

Regards,
White Rabit

**Comment by RandolphVI** [comment]
@rabitwhte 
It's a good question.
In training step, the model use the loss which consider all levels.
In validation step, the validation accuracy is calculated based on all levels, which means the thought you mentioned may exist.
But in the test step, if you are only interest in bottom level, evaluation method (see the test_harnn.py) can predict each level (including the bottom level), and each level will give corresponding evaluation score (P/R, F1).

**Comment by rabitwhte** [comment]
I just ran and went through the code of test_harnn.py and to be honest I can't really see how can you make test_harnn.py give results at each level. Right now it gives only one score (also total I think).

**Comment by RandolphVI** [comment]
@rabitwhte 

I check the version of `test_harnn.py`, it's not the latest.
I have updated the latest version, now you can check again.
